src/linear_algebra/affine_space/combination.lean

```diff
@@ -448,6 +514,40 @@ end
 
 end affine_space'
 
+section division_ring
+
+variables {k : Type*} {V : Type*} {P : Type*} [division_ring k] [add_comm_group V] [module k V]
+variables [affine_space V P] {ι : Type*}
+include V
+
+/-- The centroid lies in the affine span if the number of points,
+converted to `k`, is not zero. -/
+lemma centroid_mem_affine_span_of_cast_card_ne_zero {s : finset ι} (p : ι → P)
+  (h : (finset.card s : k) ≠ 0) : s.centroid k p ∈ affine_span k (set.range p) :=
```

Comment:
Is there a specific reason why you don't `open set finset` to make these statements shorter? If there are names clashes you can simply mark this question as resolved.

Reply:
Added `open set finset` (and also fixed up uses of `finset.` that were already unnecessary because they were inside `namespace finset`).